Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc

```diff
@@ -67,12 +67,13 @@ GEMGeometryBuilderFromDDD::build( const std::shared_ptr<GEMGeometry>& theGeometr
       GEMSuperChamber *gemSuperChamber = buildSuperChamber(fv, detIdCh);
       theGeometry->add(gemSuperChamber);
     }
-    GEMChamber *gemChamber = buildChamber(fv, detIdCh);
     
     // loop over chambers
     // only 1 chamber
     bool doChambers = fv.firstChild();
     while (doChambers){
+
+    GEMChamber *gemChamber = buildChamber(fv, detIdCh);
```

Comment:
@guitargeek - why was it moved inside the loop?

Reply:
If the loop isn't executed, the ownership of the memory is not transferred. In similar cases, I therefore introduced a bool to check if the loop executed. If not, delete manually. You are pointing out a special case here, where I took the `only 1 chamber` by word. In this case there would be no overhead, because the loop gets only executed once. But if you we can change that!

Reply:
This is a bit tricky. What is there is more then one sibling in the filtered view? 
If the loop is not executed, it means that the filtered view is empty, which means there is no GEMs to build, which means this producer should not have been called at all.
On a side note, all pointers here should be replaced by smart pointers. 
BTW, any changes to the geometry producers should be coordinated with the DPG who own the code. They will have to re-validate the geometry.

Reply:
I see. Maybe we could avoid many pointers all together? For example there is this function `GEMChamber* buildChamber()`. Wouldn't it be more idiomatic to do `GEMChamber* buildChamber()`? Since the times of return value optimization [1], this should have no overhead, right?

[1] https://en.wikipedia.org/wiki/Copy_elision#Return_value_optimization

Reply:
Ok thanks, then I try to keep the changes marginal so it doesn't have to be reevaluated. Actually the GEMGeometry was failing in the tests. I now fixed the memory leak with the simple check if the loop got executed.